packages/python-google-compute-engine/google_compute_engine/instance_setup/instance_setup.py

```diff
@@ -28,9 +28,20 @@
 from google_compute_engine import logger
 from google_compute_engine import metadata_watcher
 from google_compute_engine.boto import boto_config
+from google_compute_engine.compat import urlerror
+from google_compute_engine.compat import urlrequest
 from google_compute_engine.instance_setup import instance_config
 
 
+class PutRequest(urlrequest.Request):
+    def get_method(self):
+        return 'PUT'
+
+
+GUEST_ATTRIBUTES_URL = ('http://metadata.google.internal/computeMetadata/v1beta1/'
+                        'instance/guest-attributes/hostkeys/')
```

Comment:
Let's separate out the `hostkeys/` suffix.

Reply:
Done